fish-rust/src/common.rs

```diff
@@ -106,6 +106,16 @@ impl<T, F: FnOnce(&mut T)> Drop for ScopeGuard<T, F> {
         unsafe { ManuallyDrop::drop(&mut self.captured) };
     }
 }
+// These are in the Unicode private-use range. We really shouldn't use this
+// range but have little choice in the matter given how our lexer/parser works.
+// We can't use non-characters for these two ranges because there are only 66 of
+// them and we need at least 256 + 64.
+//
+// Note: We don't use the highest 8 bit range (0xF800 - 0xF8FF) because we know
+// of at least one use of a codepoint in that range: the Apple symbol (0xF8FF)
+// on Mac OS X. See http://www.unicode.org/faq/private_use.html.
+pub const ENCODE_DIRECT_BASE: u32 = 0xF600;
+pub const ENCODE_DIRECT_END: u32 = ENCODE_DIRECT_BASE + 256;
```

Comment:
Already exist in `src/wchar.rs`.

Reply:
Whoops, this slipped through. Fixed in b0a3e14832dfb6eb1c86618f575641a2a4e74c89. Apologies for missing this.